networkit/graph.pyx

```diff
@@ -414,6 +423,75 @@ cdef class Graph:
 
 		return self._this.addEdge(u, v, w, checkMultiEdge)
 
+	def addEdges(self, inputData, addMissing = False, checkMultiEdge = False):
+		"""
+		addEdges(inputData)
+
+		Inserts edges from several sources based on the type of :code:`inputData`.
+
+		If the graph is undirected, each pair (i,j) in :code:`inputData` is inserted twice twice: once as (i,j) and once as (j,i).
+
+		Parameter :code:`inputData` can be one of the following:
+
+		- scipy.sparse.coo_matrix
+		- (data, (i,j)) where data, i and j are of type np.ndarray
+		- (i,j) where i and j are of type np.ndarray
+
+		Note
+		----
+		If only pairs of row and column indices (i,j) are given, each edge is given weight 1.0 (even in case of a weighted graph).
+
+		Parameters
+		----------
+		inputData : several
+			Input data encoded as one of the supported formats.
+		addMissing : bool, optional
+			Add missing endpoints if necessary (i.e., increase numberOfNodes). Default: False
+		checkMultiEdge : bool, optional
+			Check if edge is already present in the graph. If detected, do not insert the edge. Default: False
+		"""
+
+		cdef cnp.ndarray[DUINT_t, ndim = 1, mode = 'c'] row, col
+		cdef cnp.ndarray[DDOUBLE_t, ndim = 1, mode = 'c'] data
+
+		if isinstance(inputData, coo_matrix):
+			try:
+				row = inputData.row.astype(np.uint).view(np.uint)
+				col = inputData.col.astype(np.uint).view(np.uint)
+				data = inputData.data.view(np.double)
+			except (TypeError, ValueError) as e:
+				raise TypeError('invalid input format') from e
+		elif isinstance(inputData, tuple) and len(inputData) == 2:
+			if isinstance(inputData[1], tuple):
+				try:
+					row = inputData[1][0].view(dtype = np.uint)
+					col = inputData[1][1].view(dtype = np.uint)
+					data = inputData[0].view(dtype = np.double)
+				except (TypeError, ValueError) as e:
+					raise TypeError('invalid input format') from e
+			else:
+				try:
+					row = inputData[0].view(dtype = np.uint)
+					col = inputData[1].view(dtype = np.uint)
+					data = np.ones(len(row), dtype = np.double)
+				except (TypeError, ValueError) as e:
+					raise TypeError('invalid input format') from e				
+		else:
+			raise TypeError('invalid input format')
+
+		cdef int numEdges = np.shape(row)[0]
+
+		if addMissing:	
```

Comment:
I am not sure why we need to call the python `addEdge` here? We could just add some code to add missing nodes instead? But I do not know the use cases for this `addMissing` parameter - some context might be useful to understand this.

Reply:
From my understanding, this is a convenient function/parameter. Basically what it does is adding the missing nodes instead (with all cases, including dynamic graphs with previously deleted nodes). I was trying to mirror the behavior of `addEdge`, which is supporting `addMissing`.

The other possibilities would be to either omit `addMissing` or re-implement the function. Do you prefer one of those?